emannotationschemas/models.py

```diff
@@ -351,125 +286,109 @@ def create_table_dict(
-            raise (InvalidTableMetaDataException(msg))
-        else:
-            try:
-                model["reference_table_name"] = table_metadata["reference_table"]
-            except KeyError:
-                msg = f"reference table not specified in metadata {table_metadata}"
-                raise InvalidTableMetaDataException(msg)
 
     for key, field in Schema._declared_fields.items():
         if not field.metadata.get("drop_column", False):
+            model = validate_metadata(model, field, table_metadata)
             model = add_column(model, key, field)
 
     return model
 
 
 def add_column(model: dict, key: str, field: str) -> dict:
     field_type = type(field)
-    do_index = field.metadata.get("index", False)
+    has_index = field.metadata.get("index", False)
     if field_type in field_column_map:
         if field_type == PostGISField:
             postgis_geom = field.metadata.get("postgis_geometry", "POINTZ")
             model[key] = Column(
-                Geometry(geometry_type=postgis_geom, dimension=3, use_N_D_index=True)
+                Geometry(
+                    geometry_type=postgis_geom, dimension=3, use_N_D_index=has_index
+                )
             )
         elif field_type == ReferenceTableField:
             reference_table_name = model.pop("reference_table_name")
             model[key] = Column(
-                BigInteger, ForeignKey(f"{reference_table_name}.id"), index=True
+                BigInteger, ForeignKey(f"{reference_table_name}.id"), index=has_index
             )
         else:
-            model[key] = Column(field_column_map[field_type], index=do_index)
+            model[key] = Column(field_column_map[field_type], index=has_index)
 
     else:
         raise InvalidSchemaField(f"field type {field_type} not supported")
 
     return model
 
 
-def make_segmentation_model_from_schema(
-    table_name: str, segmentation_source: str, Schema
-):
-    segmentation_table_name = f"{table_name}__{segmentation_source}"
-    make_annotation_model_from_schema(table_name, Schema)
-
-    if not annotation_models.contains_model(segmentation_table_name):
-        __, segmentation_columns = split_annotation_schema(Schema)
-
-        seg_model = create_segmentation_model(
-            table_name, segmentation_source, segmentation_columns
-        )
-        annotation_models.set_model(segmentation_table_name, seg_model)
-
-    return annotation_models.get_model(segmentation_table_name)
+def validate_metadata(model, field, table_metadata):
+    field_metadata = field.metadata.get("metadata")
+    if field_metadata == MetaDataTypes.REFERENCE.value:
+        if type(table_metadata) is not dict:
+            msg = "no metadata provided for reference annotation"
+            raise (InvalidTableMetaDataException(msg))
+        else:
+            try:
+                model["reference_table_name"] = table_metadata["reference_table"]
```

Comment:
i found this confusing.. model here is a dict?

Reply:
The arg should be renamed model_dict or something similar...